core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -719,54 +722,14 @@ func (vdb *VersionedDB) Close() {
 	// no need to close db since a shared couch instance is used
 }
 
-// ensureFullCommitAndRecordSavepoint flushes all the dbs (corresponding to `namespaces`) to disk
-// and Record a savepoint in the metadata db.
-// Couch parallelizes writes in cluster or sharded setup and ordering is not guaranteed.
-// Hence we need to fence the savepoint with sync. So ensure_full_commit on all updated
-// namespace DBs is called before savepoint to ensure all block writes are flushed. Savepoint
-// itself is flushed to the metadataDB.
-func (vdb *VersionedDB) ensureFullCommitAndRecordSavepoint(height *version.Height, namespaces []string) error {
-	// ensure full commit to flush all changes on updated namespaces until now to disk
-	// namespace also includes empty namespace which is nothing but metadataDB
-	errsChan := make(chan error, len(namespaces))
-	defer close(errsChan)
-	var commitWg sync.WaitGroup
-	commitWg.Add(len(namespaces))
-
-	for _, ns := range namespaces {
-		go func(ns string) {
-			defer commitWg.Done()
-			db, err := vdb.getNamespaceDBHandle(ns)
-			if err != nil {
-				errsChan <- err
-				return
-			}
-			_, err = db.ensureFullCommit()
-			if err != nil {
-				errsChan <- err
-				return
-			}
-		}(ns)
-	}
-
-	commitWg.Wait()
-
-	select {
-	case err := <-errsChan:
-		logger.Errorf("Failed to perform full commit")
-		return errors.Wrap(err, "failed to perform full commit")
-	default:
-		logger.Debugf("All changes have been flushed to the disk")
-	}
-
+// recordSavepoint records a savepoint in the metadata db for the channel.
+func (vdb *VersionedDB) recordSavepoint(height *version.Height, namespaces []string) error {
```

Comment:
`namespaces []string` is unused and not needed.

Reply:
Good catch. Let me fix this.